cmd/aws-s3-csi-controller/main.go

```diff
@@ -63,3 +69,30 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+// getClusterVariant determines Kubernetes variant by checking API groups.
+func getClusterVariant(client *rest.Config, log logr.Logger) util.ClusterVariant {
+	defaultVariant := util.DefaultKubernetes
+
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(client)
+	if err != nil {
+		log.Error(err, "Failed to create DiscoveryClient to determine cluster variant. Assuming this is Default Kubernetes variant")
+		return defaultVariant
+	}
+
+	// Get API groups
+	apiGroups, err := discoveryClient.ServerGroups()
+	if err != nil {
+		log.Error(err, "Failed to get API groups to determine cluster variant. Assuming this is Default Kubernetes variant")
+		return defaultVariant
+	}
+
+	// Check if the cluster is an OpenShift cluster by detecting the "config.openshift.io" API group
+	for _, group := range apiGroups.Groups {
+		if group.Name == "config.openshift.io" {
+			return util.OpenShift
```

Comment:
```suggestion
			log.Info("Detected OpenShift cluster variant")
			return util.OpenShift
```

Logging here that we detected OpenShift variant would be useful for debugging